### DISCUSSION TITLE
Graph: Make Fill Gradient a boolean option and use fill (0-10) to set amount

### DISCUSSION THREAD

**Discussion by howardjohn** [grafana/grafana#37014] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Right now there are two configs that can be set from 0-10: Fill and Fill Gradient

How it seems to work is:
```
If fill = 0  { no fill }
if gradient > 0 { fill gradient at configured "Fill Gradient" }
fill at configured "Fill"
```

This is confusing because changing "Fill" from 1-9 has no impact if gradient is set.

How I think it should work:

Make two configs, Fill (0-10) and Gradient (yes/no)

```
If fill = 0  { no fill }
if gradient enabled { fill gradient at configured "Fill" }
fill at configured "Fill"
```

**Why is this needed**:
To make it easier to understand and configure

**Comment by torkelo** [comment]
agree, this makes a bit more sense

**Comment by eltrufas** [comment]
Hello, I'd like to work on this issue.

**Comment by torkelo** [comment]
great! don't forget to add a dashboard schema migration for the graph model so that old dashboards are migrated to the new option model: 
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/DashboardMigrator.ts#L468

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
